benefit-finder/cypress/e2e/storybook/dataLayer.cy.js

```diff
@@ -403,18 +400,25 @@ describe('Calls to Google Analytics Object', function () {
           dataLayerValueResultsViewEligible.bfData.eligibleBenefitCount.number
         )
         .then(() => {
-          // we wait for the last event to fire
-          // eslint-disable-next-line cypress/no-unnecessary-waiting
-          cy.wait(wait).then(() => {
-            // get all the events in our layer that matches the event value
-            const ev = {
-              ...window.dataLayer.filter(
-                x => x?.event === dataLayerValueResultsViewEligible.event
-              ),
-            }
-            removeID(ev[0])
-
-            expect(ev[0]).to.deep.equal(dataLayerValueResultsViewEligible)
+          // Wait dynamically for the bf_page_change and bf_count events
+          cy.wrap(window.dataLayer).should(dataLayer => {
```

Comment:
There is quite a few of these where set up `matchingEvents` and assert `isNotEmpty`. For example, looking at lines 404-421. 

It seems like we could combine these two processes, 

for example,
```JS
        .then(() => {
          // Wait dynamically for the bf_page_change and bf_count events
          cy.wrap(window.dataLayer).should(dataLayer => {

            const matchingEvent = dataLayer.filter(
              x => x?.event === dataLayerValueResultsViewEligible.event
            )[0]

            assert.isNotEmpty(
              matchingEvent,
              'bf_page_change and bf_count events are triggered'
            )

            removeID(matchingEvent)

            expect(matchingEvent).to.deep.equal(dataLayerValueResultsViewEligible)
          })
        })
```